nncf/torch/nncf_network.py

```diff
@@ -144,9 +144,10 @@ def get_original_forward(self) -> Callable:
         Returns the forward function of the original model, unmodified by NNCF. The returned function will
         have its 0-th implicit `self` argument bound to the model object.
         """
-        if self._original_instance_forward is not None:
-            return functools.partial(self._original_instance_forward, self._model_ref)
-        return functools.partial(self._original_unbound_forward, self._model_ref)
+        forward_parameters = self._forward_signature.parameters
+        if forward_parameters and next(iter(forward_parameters)) == "self":
```

Comment:
This is not a particularly reliable way of establishing whether the `self._forward_signature` is belonged to a bound or an unbound version of the method. `self` is not a reserved name, it's just a convention-level name for the 0-th argument. You can have:
```
import torch

class MyModule(torch.nn.Module):
    def forward(self, x, y):
        return x + y

def evil_forward(self, y):
    return self - y

module = MyModule()
module.forward = evil_forward

print(module.forward(20, 21))  # prints -1
```
Was this change strictly necessary to enable multi-device model inference? Maybe the case such as above should be added to tests?

Reply:
Could you suggest an approach to handle your case. And please provide the model in which this occurs?